contracts/libs/0.8.x/MinterUtils_v0_1_1.sol

```diff
@@ -0,0 +1,276 @@
+        // split remaining funds between foundation, artist, and artist's
+        // additional payee
+        uint256 renderProviderRevenue_;
+        address payable renderProviderAddress_;
+        uint256 artistRevenue_;
+        address payable artistAddress_;
+        uint256 additionalPayeePrimaryRevenue_;
+        address payable additionalPayeePrimaryAddress_;
+        if (!isEngine) {
+            // get flagship splits
+            (
+                renderProviderRevenue_, // artblocks revenue
+                renderProviderAddress_, // artblocks address
+                artistRevenue_,
+                artistAddress_,
+                additionalPayeePrimaryRevenue_,
+                additionalPayeePrimaryAddress_
+            ) = IGenArt721CoreContractV3(genArtCoreContract)
+                .getPrimaryRevenueSplits(projectId, pricePerTokenInWei);
+        } else {
+            // get engine splits
+            uint256 platformProviderRevenue_;
+            address payable platformProviderAddress_;
+            (
+                renderProviderRevenue_,
+                renderProviderAddress_,
+                platformProviderRevenue_,
+                platformProviderAddress_,
+                artistRevenue_,
+                artistAddress_,
+                additionalPayeePrimaryRevenue_,
+                additionalPayeePrimaryAddress_
+            ) = IGenArt721CoreContractV3_Engine(genArtCoreContract)
+                .getPrimaryRevenueSplits(projectId, pricePerTokenInWei);
+            // Platform Provider payment (only possible if engine)
+            if (platformProviderRevenue_ > 0) {
+                _projectCurrency.transferFrom(
+                    msg.sender,
+                    platformProviderAddress_,
+                    platformProviderRevenue_
+                );
+            }
+        }
+        // Art Blocks payment
+        if (renderProviderRevenue_ > 0) {
+            _projectCurrency.transferFrom(
+                msg.sender,
+                renderProviderAddress_,
+                renderProviderRevenue_
+            );
+        }
+        // artist payment
+        if (artistRevenue_ > 0) {
+            _projectCurrency.transferFrom(
+                msg.sender,
+                artistAddress_,
+                artistRevenue_
+            );
+        }
+        // additional payee payment
+        if (additionalPayeePrimaryRevenue_ > 0) {
+            _projectCurrency.transferFrom(
+                msg.sender,
+                additionalPayeePrimaryAddress_,
+                additionalPayeePrimaryRevenue_
+            );
+        }
+    }
+
+    function getV3CoreIsEngine(
+        IGenArt721CoreContractV3_Base genArt721CoreV3_Base
+    ) internal returns (bool isEngine) {
+        // determine if core is engine based off of coreType() response
+        isEngine =
+            keccak256(abi.encodePacked(genArt721CoreV3_Base.coreType())) ==
+            keccak256(abi.encodePacked(FLAGSHIP_CORE_TYPE));
```

Comment:
I wonder if we should check types against all expected types (rather than just checking flagship)–and throw if we hit an unexpected type?

Reply:
I think that would be ideal, but I didn't want to constrain our minters to a set of engine types in case we add more in the future. Right now, we have flagship and explorations that return three payees, and both of those (intentionally) return the same coreType. When we add EngineFlex in the future, this will automatically work.

Another option here is to simply skip the coreType check altogether, and just define isEngine based on the length of the bytes returned from the get primary splits function.

Reply:
> Another option here is to simply skip the coreType check altogether, and just define isEngine based on the length of the bytes returned from the get primary splits function.

This def feels the move.

Reply:
I'll refactor this to look only at the returned bytes, per our discussion 🎉

Reply:
Refactor has been pushed 🥳

Reply:
Wooo–thank you ser :)

Reply:
resolving this based on subsequent review feedback 🙏